### PR TITLE
Retry expired logins in notebooks

### PR DESCRIPTION
**Pull request by Sumukh**
I'm not sure how I got into a situation where my token couldn't be refreshed - but if someone does we should have them auth again. 

image

I only put it in the notebook code because it immediately accesses the email on every call to auth - unlike the normal auth flow.

### CODE REVIEW

File: client/utils/auth.py [hunk 1]
```diff
@@ -318,13 +327,13 @@ def get_info(assignment, access_token):
...
 
-def display_student_email(assignment, access_token):
+def display_student_email(assignment, access_token, warn=True):
     try:
         email = get_info(assignment, access_token)['email']
         print('Successfully logged in as', email)
         return email
-    except Exception:
-        log.warning('Could not get student email', exc_info=True)
+    except:
```
Comment (knrafto) [inline client/utils/auth.py]: Why the change from `Exception` to catching anything? That means e.g. keyboard interrupts will be swallowed

File: client/utils/auth.py [hunk 2]
```diff
@@ -318,13 +327,13 @@ def get_info(assignment, access_token):
     response.raise_for_status()
     return response.json()['data']
 
-def display_student_email(assignment, access_token):
+def display_student_email(assignment, access_token, warn=True):
```
Comment (knrafto) [inline client/utils/auth.py]: `warn` is unused

**Comment by Sumukh** [comment]
Tested by corrupting my access_token and refresh_token and it behaved as expected. Will merge after tests complete.
